### ISSUE TITLE
关于example中生成类模型token_type_ids 的padding问题和mask问题

### ISSUE THREAD

**Issue by siyu1992** [920232796/bert_seq2seq#20]
您好，非常感谢您贡献了这么好的repo给我们试用和学习，有两个问题想请教一下，还请帮忙指点指点：

1.关于example中给出的训练代码，像写诗这样的需要生成序列的任务中，我观察到collate_fn这个函数中的token_ids_padded和token_type_ids_padded调用的padding方法中，padding试用的id是0，但是对于输入中有2个序列的情况，如果padding id是0，不就是说补的padding token是属于第一个sequence了吗？但其实位置是在第二个sequence的，这个会不会对模型训练造成影响呢？
代码如下：
def collate_fn(batch):
    """
    动态padding， batch为一部分sample
    """

    def padding(indice, max_length, pad_idx=0):
        """
        pad 函数
        """
        pad_indice = [item + [pad_idx] * max(0, max_length - len(item)) for item in indice]
        return torch.tensor(pad_indice)

    token_ids = [data["token_ids"] for data in batch]
    max_length = max([len(t) for t in token_ids])
    token_type_ids = [data["token_type_ids"] for data in batch]

    token_ids_padded = padding(token_ids, max_length)
    token_type_ids_padded = padding(token_type_ids, max_length)
    target_ids_padded = token_ids_padded[:, 1:].contiguous()

    return token_ids_padded, token_type_ids_padded, target_ids_padded

我个人认为   token_type_ids_padded = padding(token_type_ids, max_length)需要传一个pad_idx=1参数，请问下这样是否才正确，不对也请帮忙指正

2.Seq2SeqModel中forward函数中，构建特殊的mask这里，s_ex12 和s_ex13这两个mask和a_mask = (1.0 - s_ex12) * (1.0 - s_ex13) + s_ex13 * a_mask的计算方法不是很理解，可否帮忙指点或者提供一点参考链接？

感激不尽

**Comment by 920232796** [comment]
1. 我个人觉得pad部分无所谓，因为不管token-type是什么，有什么含义，但是因为是pad部分，模型里面会把这部分的输出变为负很大的数，因此softmax以后注意力影响基本为0。

**Comment by 920232796** [comment]
2. 这部分，我刚开始写这个的时候，好像是用for循环+torch的一些函数做的，跑起来非常慢。。但是看了苏剑林老师的代码以后，才发现这样运算可以很好的构造mask，于是就模仿了一下下～ 哈哈 你可以去看下苏剑林老师的代码。

**Comment by siyu1992** [comment]
可以给个链接吗，我也想去学习一下，谢谢啦

**Comment by 920232796** [comment]
我刚看了一下，由于更新了很多版了，他好像换方法了，你可以看下新的方法：https://github.com/bojone/bert4keras/blob/master/bert4keras/models.py
原理的话其实感觉你可以自己写一个简单的输入，然后带入函数试一下～就能明白了。

**Comment by siyu1992** [comment]
收到，感谢感谢
